src/GraphUtils.ts

```diff
@@ -426,16 +430,31 @@ export function findTurnPts(pts: Point[], mode: string, isClosed: boolean = fals
         if (pts[i].y > pts[mod(i-1, pts.length)].y) {
             while (pts[i].y == pts[mod(i+1, pts.length)].y) {
                 i += 1;
+                if (i == pts.length - 1) {
+                    return false;
+                }
             }
             return pts[i].y > pts[mod(i+1, pts.length)].y;
         }
         return false;
     };
 
+    // does any point within 5 points of i have a y value greater than (maxima) / less than (minima) (remember y is inverted) EXTREMA_DETECTION_HEIGHT units further than pts[i].y?
+    const isNotable = (i: number, style: TurningPointStyle) => {
+        for (let j = isClosed ? 0 : Math.min(0, i - 5); j < (isClosed ? pts.length : Math.max(pts.length, i + 5)); j++) {
```

Comment:
This currently searches through all points rather than the local neighbourhood
```suggestion
        for (let j = isClosed ? mod(i - 5, pts.length) : Math.max(0, i - 5); j < (isClosed ? mod(i + 5, pts.length) : Math.min(pts.length, i + 5)); j++) {
```